go/fn/object_test.go

```diff
@@ -10,6 +10,78 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+type GVKTest struct {
+	group   string
+	version string
+	kind    string
+	resultA bool
+	resultB bool
+	resultC bool
+}
+
+func TestGVK(t *testing.T) {
+	input := []byte(`
+apiVersion: apps/v3
+kind: StatefulSet
+metadata:
+  name: my-config
+spec:
+  volumeClaimTemplates:
+    - metadata:
+        labels:
+          testkey: testvalue
+`)
+
+	input_noGroup := []byte(`
+apiVersion: v3
+kind: StatefulSet
+metadata:
+  name: my-config
+spec:
+  volumeClaimTemplates:
+    - metadata:
+        labels:
+          testkey: testvalue
+`)
+
+	input_differentKind := []byte(`
+apiVersion: apps/v1
+kind: Service
+metadata:
+  name: my-config
+spec:
+  volumeClaimTemplates:
+    - metadata:
+        labels:
+          testkey: testvalue
+`)
+	testcases := map[string]GVKTest{}
+	testcases["1"] = GVKTest{"apps", "", "StatefulSet", true, true, false}
+	testcases["2"] = GVKTest{"", "v3", "StatefulSet", true, true, false}
+	testcases["3"] = GVKTest{"apps", "v3", "StatefulSet", true, true, false}
+	testcases["4"] = GVKTest{"", "", "StatefulSet", true, true, false}
+	testcases["5"] = GVKTest{"", "", "", true, true, true}
+	testcases["6"] = GVKTest{"appWrong", "", "", false, true, false}
+	testcases["7"] = GVKTest{"", "", "Service", false, false, true}
+	testcases["8"] = GVKTest{"", "v1", "", false, false, true}
+
+	resource, _ := ParseKubeObject(input)
+	resource_noGroup, _ := ParseKubeObject(input_noGroup)
+	resource_diffKind, _ := ParseKubeObject(input_differentKind)
+	for testname, data := range testcases {
+		if resource.IsGVK(data.group, data.version, data.kind) != data.resultA {
+			t.Errorf("Test case " + testname + " resource failed for isGVK")
+		}
+		if resource_noGroup.IsGVK(data.group, data.version, data.kind) != data.resultB {
+			t.Errorf("Test case " + testname + " resource_noGroup failed for isGVK")
+		}
+		if resource_diffKind.IsGVK(data.group, data.version, data.kind) != data.resultC {
+			t.Errorf("Test case " + testname + " resource_diffKind failed for isGVK")
```

Comment:
I suggest using  the assert package. The import path you can add to the top of the file is `github.com/stretchr/testify/assert`.

Then here you can do something like `assert.Equal(t, resource_diffKind.IsGVK(data.group, data.version, data.kind), data.resultC)` and it will give a very descriptive error message for you.